src/main/java/movie/storage/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package movie.storage.dao.impl;
+
+import java.util.Optional;
+import movie.storage.dao.UserDao;
+import movie.storage.exception.IncorrectDataException;
+import movie.storage.lib.Dao;
+import movie.storage.model.User;
+import movie.storage.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new IncorrectDataException("Unable to add user" + user, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Unable to add user" + user, e);
```
this will suit here better

Reply:
Done.